rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/HelmBakeTemplateUtils.java

```diff
@@ -103,4 +111,34 @@ protected Path getHelmTypePathFromArtifact(
 
     return helmTypeFilePath;
   }
+
+  /**
+   * This is a helper method to build the appropriate overrides in the event that an
+   * ArtifactReferenceURI was passed in an override
+   */
+  protected List<String> buildOverrideList(Map<String, Object> overrides) {
+    return overrides.entrySet().stream()
+        .map(
+            entry ->
+                entry.getKey() + "=" + expandArtifactReferenceURIs(entry.getValue()).toString())
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * In the event that we encounter and ArtifactReferenceURI, we want to pull down that artifact
+   * instead of using the raw URI as a value for helm.
+   */
+  private Object expandArtifactReferenceURIs(Object value) {
+    if (artifactStore == null || !(helmConfig.isExpandOverrides() && value instanceof String)) {
+      return value;
+    }
+
+    String ref = (String) value;
+    if (ArtifactReferenceURI.is(ref)) {
+      Artifact artifact = artifactStore.get(ArtifactReferenceURI.parse(ref));
+      return artifact.getReference();
```

Comment:
Tempting to add a log message here, even at debug, so there's some evidence that this is happening.  I suppose [this one](https://github.com/spinnaker/rosco/blob/be3e7923f837d12c434e0b80a9d2fe15ba75351d/rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/jobs/local/JobExecutorLocal.groovy#L55) is sufficient though.